application/common/models/Mfa.php

```diff
@@ -205,14 +227,30 @@ protected function hasTooManyRecentFailures(): bool
 
     /**
      * @param string|array $value
+     * @param string $rpOrigin Optional
+     * @param string $verifyType Optional. If not blank, it must be 'registration', referring to verifying a webauthn registration
      * @return bool
      * @throws ServerErrorHttpException
      * @throws TooManyRequestsHttpException
      * @throws \Throwable
      * @throws \yii\db\StaleObjectException
      */
-    public function verify($value, string $rpOrigin = ''): bool
+    public function verify($value, string $rpOrigin = '', string $verifyType = ''): bool
     {
+        if ($verifyType != "") {
+            if ($this->type != self::TYPE_WEBAUTHN) {
+                throw new BadRequestHttpException(
+                    'A non-blank verification type is not allowed when verifying a mfa of type ' . $this->type
+                );
+            }
+
+            if ($verifyType != Mfa::VERIFY_REGISTRATION) {
+                throw new BadRequestHttpException(
+                    'A non-blank verification type for a ' . self::TYPE_WEBAUTHN . " may only be: " . self::VERIFY_REGISTRATION
```

Comment:
I'm a bit uncomfortable with using a string like a boolean. I might suggest using an actual `bool` or using two separate, non-empty strings for the two values.

Reply:
@forevermatt Do you have a preference?  I guess I could have `actionVerify` convert the path parameter to a boolean with something like `verifyingRegistration = ($type == "registration")`

Is there a better way to do it?

Reply:
@forevermatt Do you have a preference? I guess I could have actionVerify convert the path parameter to a boolean with something like verifyingRegistration = ($type == "registration")

Is there a better way to do it? (Is there a way to have a proper boolean come through in a url path with Yii2 routes?)